### PR TITLE
Recalculate font-size when generic changes

### PR DESCRIPTION
**Pull request by Manishearth**
With the MathML refactorings this feature got lost. It would still get
recalculated when explicitly specified as a keyword, but not otherwise.

To avoid hitting the font metrics provider too often, we only do this
when the generic changes. Otherwise we trust the existing calculated
font.

I swear, once Stylo lands I'm going to campaign to remove font-size from CSS entirely. :weary:

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16824)

### CODE REVIEW

File: components/style/properties/longhand/font.mako.rs [hunk 1]
```diff
@@ -914,6 +914,18 @@ ${helpers.single_keyword_system("font-variant-caps",
...
+        if context.style().get_font().gecko().mLanguage.mRawPtr !=
+           context.inherited_style().get_font().gecko().mLanguage.mRawPtr ||
+           context.style().get_font().gecko().mGenericID !=
+           context.inherited_style().get_font().gecko().mGenericID {
+            if let Some((kw, ratio)) = context.style().font_size_keyword {
+                computed = kw.to_computed_value(context).scale_by(ratio);
+            }
+        }
+
+
```
Comment (heycam) [inline components/style/properties/longhand/font.mako.rs]: Nit: probably don't need two blank lines.

**Review by heycam — APPROVED** [review]

**Comment by jdm** [comment]
```
error: no field `mRawPtr` on type `gecko_bindings::structs::root::nsCOMPtr`
     --> /home/servo/buildbot/slave/linux-rel-css/build/target/geckolib/release/build/style-d2933bc67d5faebe/out/properties.rs:26924:61
      |
26924 |             if context.style().get_font().gecko().mLanguage.mRawPtr !=
      |                                                             ^^^^^^^ did you mean `_base`?

error: no field `mRawPtr` on type `gecko_bindings::structs::root::nsCOMPtr`
     --> /home/servo/buildbot/slave/linux-rel-css/build/target/geckolib/release/build/style-d2933bc67d5faebe/out/properties.rs:26925:71
      |
26925 |                context.inherited_style().get_font().gecko().mLanguage.mRawPtr ||
      |                                                                       ^^^^^^^ did you mean `_base`?

error: aborting due to 2 previous errors

error: Could not compile `style`.
```

**Comment by Manishearth** [comment]
@bors-servo r=heycam

nscomptr is different on debug and release :|

**Pull request merged**